src/pb_stub.cc

```diff
@@ -1101,13 +1106,76 @@ PYBIND11_EMBEDDED_MODULE(c_python_backend_utils, module)
             py::object loop =
                 py::module_::import("asyncio").attr("get_running_loop")();
             py::cpp_function callback = [infer_request]() {
-              auto response = infer_request->Exec();
-              return response;
+              auto responses = infer_request->Exec(false /* is_stream*/);
+              return responses[0];
             };
             py::object future =
                 loop.attr("run_in_executor")(py::none(), callback);
             return future;
           })
+      .def(
+          "stream_exec",
+          [](std::shared_ptr<InferRequest>& infer_request,
+             const int32_t execution_timeout) {
+            std::mutex timeout_mu;
+            std::condition_variable timeout_cv;
+            std::vector<std::shared_ptr<InferResponse>> responses;
+            std::thread t([&timeout_cv, &responses, &infer_request]() {
```

Comment:
Spawning a thread for each request is a lot of overhead. I think it would be best to just use the server timeout.